py/terra_widgets/workspace_paths.py

```diff
@@ -0,0 +1,104 @@
+"""Methods to obtains paths to files within the workspace bucket."""
+
+import datetime
+import fnmatch
+import os
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+
+
+WorkspaceDestination = NamedTuple('WorkspaceDestination', [('html_file', str), ('comment_file', str)])
+
+
+class WorkspacePaths:
+  """Encapsulate all logic for manipulating workspace paths.
+
+  Paths are of the form:
+    gs://fc-secure-<guid>/reports/<user>@researchallofus.org/<date>/<time>/<notebook>.html
```

Comment:
nit: the path prefix has changed in the past, I would treat the whole bucket name as opaque when describing the structure.

Reply:
Fixed the comments (the code was not actually looking for that prefix).